catkin_ws/src/vision/src/object_detection_utils.py

```diff
@@ -66,6 +70,7 @@ def cleanPointCloud(self, point_cloud):
         point_cloud[far_mask] = np.array([np.nan, np.nan, np.nan])
         return point_cloud
     def getPointCloud(self, bbox=None):
+        updatePointCloud()
```

Comment:
I don't think this works here for 2 reasons: since we never actually do any front cam detection until a point cloud exists (is not None), we'll never get to the point in the code where getPointCloud() is called. It'll be a sort of deadlock state where object_detection.py is waiting for the State class to have a non-None point cloud attribute, but the State class only has a non-None point cloud attribute when it's getPointCloud() function is called, and its only called after object_detection.py proceeds with a detection. Also, the same problem would happen with the pausing: getPointCloud() should only be called after the State class is paused, which means that the point cloud will never actually update since it's always paused

Reply:
I think it might be cleaner and avoid these problems by creating a subscriber inside the State class for image and camera_info topics and using the functions from point_cloud.py to turn them into a usable point cloud array. It would also avoid having to use global variables for depth